src/compiler/scala/tools/nsc/typechecker/SyntheticMethods.scala

```diff
@@ -326,14 +326,15 @@ trait SyntheticMethods extends ast.TreeDSL {
 
     /* If you serialize a singleton and then deserialize it twice,
      * you will have two instances of your singleton unless you implement
-     * readResolve.  Here it is implemented for all objects which have
-     * no implementation and which are marked serializable (which is true
-     * for all case objects.)
+     * readResolve.  Here we use a serialization proxy that prevents
```

Comment:
I originally wrote this confusing comment. I think @mkeskells' version in https://github.com/scala/scala/pull/6877/files#r199624245 is better.